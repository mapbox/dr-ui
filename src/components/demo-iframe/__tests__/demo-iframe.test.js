import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './demo-iframe-test-cases.js';

describe('demo-iframe', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.token.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.token;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.missing.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.missing;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toBe(null);
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.nogl.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.nogl;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree.children[0].type).toBe('iframe');
      expect(tree).toMatchSnapshot();
    });
  });
});
