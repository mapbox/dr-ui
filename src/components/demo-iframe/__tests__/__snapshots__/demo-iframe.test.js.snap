// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demo-iframe \`gl\` is false renders as expected 1`] = `
<div>
  <iframe
    height="400px"
    src="https://giphy.com/embed/ule4vhcY1xEKQ"
    width="100%"
  />
  <a
    className="link"
    href="https://giphy.com/embed/ule4vhcY1xEKQ"
  >
    <span>
      View fullscreen
       
      <span
        className="txt-nowrap"
      >
        demo
        <svg
          className="events-none icon inline-block align-t"
          focusable="false"
          role="presentation"
          style={
            Object {
              "height": "1.5em",
              "width": "1.5em",
            }
          }
        >
          <use
            xlinkHref="#icon-chevron-right"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </span>
    </span>
  </a>
</div>
`;

exports[`demo-iframe Basic renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#feefe2",
      "color": "#945823",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-orange-light round-full color-orange flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": "40",
            "width": "40",
          }
        }
      >
        <use
          xlinkHref="#icon-alert"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Mapbox GL unsupported
    </div>
    Mapbox GL requires
     
    <a
      className="link"
      href="https://caniuse.com/#feat=webgl"
    >
      WebGL support
    </a>
    . Please check that you are using a supported browser and that
     
    <a
      className="link"
      href="https://get.webgl.org/"
    >
      WebGL is enabled
    </a>
    .
  </div>
</div>
`;

exports[`demo-iframe Missing token will return null component renders as expected 1`] = `null`;

exports[`demo-iframe With token renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#feefe2",
      "color": "#945823",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-orange-light round-full color-orange flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": "40",
            "width": "40",
          }
        }
      >
        <use
          xlinkHref="#icon-alert"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Mapbox GL unsupported
    </div>
    Mapbox GL requires
     
    <a
      className="link"
      href="https://caniuse.com/#feat=webgl"
    >
      WebGL support
    </a>
    . Please check that you are using a supported browser and that
     
    <a
      className="link"
      href="https://get.webgl.org/"
    >
      WebGL is enabled
    </a>
    .
  </div>
</div>
`;
